blockchain-proxy: Add `Light` variant

jsdanielh opened the pull request:
- Add the `Light` variant of the `BlockchainProxy`. This allows to instantiate a `NanoBlockchain` by using the correspondent variant of the `BlockchainProxy`.
- Rename `NanoBlockchain` to `LightBlockchain` and also renamed the crate name from `nano-blockchain` to `light-blockchain.
  Also change its references accordingly.

## Pull request checklist

- [X] All tests pass. Demo project builds and runs.
- [X] I have resolved any merge conflicts.

jsdanielh committed:
blockchain-proxy: Add `Light` variant

Add the `Light` variant of the `BlockchainProxy`. This allows to
instantiate a `NanoBlockchain` by using the correspondent variant of
the `BlockchainProxy`.

paberr reviewed (approved):
Looks good.
Having in mind that we want to slowly change names from nano to light, we could already do that for the NanoBlockchain.
But better check with @viquezclaudio whether he already is working with it to avoid conflicts.

jsdanielh committed:
Rename `NanoBlockchain` to `LightBlockchain`

Rename `NanoBlockchain` to `LightBlockchain` and also renamed the
crate name from `nano-blockchain` to `light-blockchain.
Change its references accordingly.

jsdanielh commented:
> Looks good. Having in mind that we want to slowly change names from nano to light, we could already do that for the NanoBlockchain. But better check with @viquezclaudio whether he already is working with it to avoid conflicts.

Done renaming `NanoBlockchain` to `LightBlockchain`. @paberr @viquezclaudio

viquezclaudio commented:
LGTM
